### PR TITLE
simplfying json serialization and deserialization, manager for json c…

### PR DESCRIPTION
**Pull request by igasm**
I simplify classes responsible for json conversion, and also add manager for conversion.

### CODE REVIEW

File: common/src/main/java/battleships/communication/MessageManager.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package battleships.communication;
+
+import java.util.Optional;
+
+public interface MessageManager {
+    public String toStringMessage(Messagable messegable);
```
Comment (robwrobel) [inline common/src/main/java/battleships/communication/MessageManager.java]: toString

File: common/src/main/java/battleships/communication/MessageManager.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    public Optional<Messagable> fromStringMessage(String message);
```
Comment (robwrobel) [inline common/src/main/java/battleships/communication/MessageManager.java]: toMessagable

File: common/src/main/java/battleships/communication/MessageManager.java (hunk 1 above)
Comment (igasm) [reply common/src/main/java/battleships/communication/MessageManager.java]: done

File: common/src/main/java/battleships/communication/MessageManager.java (hunk 2 above)
Comment (igasm) [reply common/src/main/java/battleships/communication/MessageManager.java]: done

**Comment by igasm** [comment]
I added changes as Robert request.

**Pull request merged**